src/librbd/AioImageRequest.cc

```diff
@@ -382,6 +390,107 @@ void AioImageWrite::update_stats(size_t length) {
   m_image_ctx.perfcounter->inc(l_librbd_wr_bytes, length);
 }
 
+void AioImageWriteSame::assemble_extent(const ObjectExtent &object_extent,
+                                    bufferlist *bl) {
+  if (object_extent.offset % m_data_len == 0
```

Comment:
Am I reading this incorrectly or is there an assumption that the offset should always be a multiple of the data length?